### PR TITLE
RI-8223 Per-element array delete in View and Search tabs

### PR DESCRIPTION
**Pull request by VaskoAtanasovRedis**
# What

UI of the array **Delete** vertical (RI-8223) — slice **S2a: per-element delete**. Adds a trash-can affordance on every populated row of **both the View tab and the Search-results tab**, wired to `ARDEL` (the `DELETE /array/elements` endpoint from the stacked S1 PR #6133).

- New `deleteArrayElements` thunk deletes by index, then re-checks `ARCOUNT`: if elements remain it refreshes the count + key-info and toasts "removed element"; if the key is now gone (last element deleted) it removes the key from the list and toasts "deleted key" (no error toast). No client-side count math, so no u64 rounding.
- `ArrayDetailsTable` gains an **opt-in** actions column (`RowActionsCell` + the shared `PopoverDelete` confirmation). A shared `useArrayElementActions` hook owns the popover state; each tab passes its own post-delete refresh — View re-runs the range/scan query, Search re-runs the search.
- The trash icon shows only on **real, deletable rows**: hidden on empty View gaps (`value == null`), but kept on Search index-only matches (WITHVALUES off, where a real element also has a null value).

**Stacked on #6133** (S1 backend) — review/merge that first; the base branch is `be/RI-8223/array-delete-endpoints`. **Out of scope:** multi-select bulk delete (S2b) and range delete `ARDELRANGE` (S3).

# Testing

- `yarn test` — array slice + all `array-details` specs (incl. the `ArrayDetails` integration) pass: thunk element-removed vs last-element-removed paths, opt-in actions column, populated-only icon, **Search index-only matches stay deletable**, and per-element delete dispatching `ARDEL` with the right index from both tabs.
- `yarn lint` clean; UI `yarn type-check` adds no new errors.
- Local Codex pre-push review run — it surfaced the Search WITHVALUES-off regression, which is fixed here before pushing.

https://github.com/user-attachments/assets/c841e3eb-6481-4e83-935f-0ea7d5aa6dd8

---

Part of RI-8223

---

> [!NOTE]
> **Medium Risk**
> Deletes live Redis array elements and can remove keys from the browser when the last element is deleted; mitigated by confirm popovers, selection guards, and patterns aligned with list element delete.
> 
> **Overview**
> Adds **per-element delete** for array keys on the **View** and **Search** tabs, wired to `DELETE array/elements` (ARDEL) via a new `ARRAY_ELEMENTS` API constant and `deleteArrayElements` thunk.
> 
> **`ArrayDetailsTable`** gains an opt-in actions column: **`RowActionsCell`** uses the shared **`PopoverDelete`** confirm flow; trash icons show on row hover/focus. **`useArrayElementActions`** owns popover state and dispatches the thunk. View passes `hideEmptySlots: true` (no delete on gap rows); Search uses `hideEmptySlots: false` so index-only matches with null values stay deletable.
> 
> After a successful delete, the thunk probes **`ARCOUNT`**: 404 means the key was removed (last element) → key list + deleted-key toast; otherwise **`refreshArray`** replays View, Search, and Aggregate plus key info, with success toasts. Mid-flight **instance/key** changes skip UI updates (app-context selection + byte-safe key compare). **`ARRAY_ELEMENT_DELETED`** telemetry is emitted.
> 
> Aggregate tab stays read-only (no `deleteConfig`). Tests cover thunk edge cases and tab-level delete UX.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 17ab7285e2a2ea9860dd6271076b067553db1878. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: redisinsight/ui/src/slices/browser/array.ts [hunk 1]
```diff
@@ -538,6 +550,75 @@ export function fetchArrayCount(key: RedisString) {
...
+      } catch (countError) {
+        // 404 ⇒ the last element went and the server dropped the key. Any
+        // other ARCOUNT failure is unrelated to the (already-succeeded) delete,
+        // so fall through and refresh instead of masking it as a failure.
+        const countStatus = get(countError, ['response', 'status'])
+        keyDeleted = Boolean(countStatus && isStatusNotFoundError(countStatus))
+      }
+
+      if (keyDeleted) {
+        dispatch(deleteSelectedKeySuccess())
```
[the remark this answers is not in the document]
Comment (VaskoAtanasovRedis) [reply redisinsight/ui/src/slices/browser/array.ts:612]: Addressed. `deleteArrayElements` captures `startInstanceId` up front and, after the DELETE + ARCOUNT round-trip, re-reads state and returns before dispatching unless the connected instance still matches and the current selection still byte-equals `key` (`isEqualBuffers`). A mid-flight key switch therefore no longer clears the newly-selected key or refreshes the shared array slice/header with the old key. Resolving.

File: redisinsight/ui/src/slices/browser/array.ts [hunk 2]
```diff
@@ -538,6 +552,90 @@ export function fetchArrayCount(key: RedisString) {
...
+        const countStatus = get(countError, ['response', 'status'])
+        keyDeleted = Boolean(countStatus && isStatusNotFoundError(countStatus))
+      }
+
+      // If the user switched database or array key while the delete was in
+      // flight, the shared array slice/header now belong to a different
+      // selection. The delete already applied server-side, so skip the UI
+      // updates rather than clobber the current view with the old key's data.
+      const latest = stateInit()
+      const selectedKey = selectedKeyDataSelector(latest)?.name
```
[the remark this answers is not in the document]
Comment (VaskoAtanasovRedis) [reply redisinsight/ui/src/slices/browser/array.ts]: Fixed in 3249e9787: the guard now reads the app-context selection (`appContextSelectedKey`, set synchronously on key click) instead of `selectedKey.data`, which lagged behind a key switch while the new key info loaded. Added a regression test that fails against the old `data`-based guard. Resolving.

File: redisinsight/ui/src/slices/browser/array.ts [hunk 3]
```diff
@@ -538,6 +552,94 @@ export function fetchArrayCount(key: RedisString) {
...
+      // selection. The delete already applied server-side, so skip the UI
+      // updates rather than clobber the current view with the old key's data.
+      const latest = stateInit()
+      // Read the user's current selection, not `selectedKey.data`: on a key
+      // switch the details reducer keeps the old `data` in place while the new
+      // key info loads, so `data.name` would still return the deleted key
+      // during that window and the guard below would wrongly pass.
+      const selectedKey = appContextSelectedKey(latest)
+      const sameInstance =
+        latest.connections.instances.connectedInstance?.id === startInstanceId
```
[the remark this answers is not in the document]
Comment (VaskoAtanasovRedis) [reply redisinsight/ui/src/slices/browser/array.ts:604]: Guarded in practice. `BrowserPage.onChangeDbIndex` clears the selected key on any DB-index switch, and the stale-delete guard bails unless the current app-context selection still byte-equals `key`. So the moment the DB changes, the selection is cleared and the ARCOUNT probe/refresh are skipped. The residual window this describes needs the user to switch DB *and* re-select a same-named key in the new DB within the DELETE+ARCOUNT round-trip (~100ms) - two deliberate actions inside a sub-100ms window. Not worth a third guard dimension (and a redux DB compare wouldn't perfectly match the per-request `ri-db-index` header the request actually uses). Resolving as negligible.

**Review by ArtemHoruzhenko — DISMISSED** [review]
lgtm

**Review by pawelangelow — DISMISSED** [review]

**Review by ArtemHoruzhenko — APPROVED** [review]

File: redisinsight/ui/src/slices/browser/array.ts [hunk 4]
```diff
@@ -538,6 +552,94 @@ export function fetchArrayCount(key: RedisString) {
...
+// Per-element delete (ARDEL). The slice's data.count isn't kept fresh and
+// arrays are sparse, so probe ARCOUNT after the delete to learn whether the
+// key survived: it 404s once the last element is gone (the server drops the
+// empty key). Any other ARCOUNT outcome leaves the delete intact, so the views
+// are refreshed regardless rather than reporting the delete as failed.
+export function deleteArrayElements(key: RedisString, indexes: string[]) {
+  return async (dispatch: AppDispatch, stateInit: () => RootState) => {
+    try {
+      const state = stateInit()
+      const startInstanceId = state.connections.instances.connectedInstance?.id
```
[the remark this answers is not in the document]
Comment (VaskoAtanasovRedis) [reply redisinsight/ui/src/slices/browser/array.ts:564]: Agreed the DB index isn't in the guard - this is the same mid-flight-DB-switch race as the earlier thread, and the assessment holds. It requires a same-connection DB switch plus re-selecting a byte-identical key name in the new DB, all inside the ~100ms DELETE+ARCOUNT round-trip. The DELETE already applied to the correct DB, so the worst case is a misleading toast / a refresh of the same-named key in the new DB - no data loss and nothing deleted in the wrong place. For a window this narrow we're accepting it rather than adding a third guard dimension. If it ever proves reachable in practice, the clean fix is to capture the session-storage db-index (the same source the request interceptor uses, keyed by instance id) at thunk start and require it to still match. Resolving as negligible for now.

**Review by pawelangelow — APPROVED** [review]
